src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,18 +1,21 @@
 package org.scalameter
 
 
-import collection._
-import compat._
-import utils.{withGCNotification, Tree}
+import utils.Tree
 
 
 
 trait Executor {
 
+  def measurer: Measurer
+
   def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+    val context = measurer.prepareContext(setuptree.context)
```

Comment:
Json based can easily bypass it.
Serialization-based persistor could have problems with it however. We
could address those in a separate PR.

On 7/9/2015 1:51 AM, Krzysztof Janosz wrote:
>>    def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
>> +    val context = measurer.prepareContext(setuptree.context)
> 
> Also, about skipping some keys - it should be bypassed in all persistors? Or only in json based? 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/scalameter/scalameter/pull/135/files#r34211586
> 

-- 
Aleksandar Prokopec